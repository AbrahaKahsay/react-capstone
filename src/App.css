body {
    font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
}

p {
    font-family: 'Lato';
}

h2 {
    font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
}