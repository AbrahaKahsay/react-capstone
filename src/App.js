import './App.css';
import React from 'react'
import { BrowserRouter, Routes, Route } from 'react-router-dom';

function App() {
  return (
    <>
    <BrowserRouter>
      <Navbar />
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/detail/:id" element={<Detail />} />
      </Routes>
    </BrowserRouter>
    </>
  );
}

export default App;
