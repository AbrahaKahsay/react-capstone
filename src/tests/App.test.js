import renderer from 'react-test-renderer';
import { render, screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import App from '../App';

describe('Jest Snapshot testing suite', () => {
  it('Check the Crypto Market', async () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    await waitFor(() => {
      const waitForDOM = screen.findByText(
        'Crypto Market',
      );
      expect(waitForDOM).not.toBeNull();
    });
  });

  it('It should match snapshot', () => {
    const Tree = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    ).toJSON();
    expect(Tree).toMatchSnapshot();
  });
});
