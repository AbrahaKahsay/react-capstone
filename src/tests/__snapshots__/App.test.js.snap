// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite It should match snapshot 1`] = `
Array [
  <nav
    className="nav-container"
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="home"
        className="icons"
        src="home.png"
      />
    </a>
    <div
      className="back"
    >
      <h2>
        Crypto Market
      </h2>
    </div>
    <div
      className="mic-set"
    >
      <img
        alt="mic"
        className="icons"
        src="mic.png"
      />
      <img
        alt="settings"
        className="icons"
        src="settings.png"
      />
    </div>
  </nav>,
  <section>
    <div
      className="info-list"
    >
      <div
        className="options"
      >
        <select
          className="filter"
          onChange={[Function]}
        >
          <option
            value="price"
          >
             Rank 
          </option>
          <option
            value="name"
          >
             Name 
          </option>
          <option
            value="vol"
          >
             Price 
          </option>
        </select>
      </div>
      <div
        className="forms"
      >
        <form
          className="search-form"
          onSubmit={[Function]}
        >
          <input
            name="coinSearch"
            placeholder="Fill Crypto name ..."
            required={true}
            type="text"
          />
          <input
            type="submit"
            value="Search"
          />
        </form>
      </div>
    </div>
    <ul
      className="coinList"
    />
  </section>,
]
`;
